docs/ota-client-guide/modules/ROOT/pages/virtual-secondaries.adoc

```diff
@@ -0,0 +1,45 @@
+= Use of Virtual Secondaries
+ifdef::env-github[]
+
+[NOTE]
+====
+We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
+====
+endif::[]
+
+
+Virtual secondaries can be used as a way to update a configuration file on the primary (where the primary is a standard OSTree image running an aktualizr). You can do this in two parts:
+
+* First you need a virtual secondary configured that can be used to update a configuration file.
+* Afterwards you can integrate it into your Yocto build.
+
+
+
+== Creating a Virtual Secondary
+
+. You can create a virtual secondary to run locally by following the steps on how to xref:simulate-device-basic.adoc[simulate a device without building a disk image].
++
+You can use the provided `.deb` releases mentioned and the `virtualsec.json` file, which you can edit to suit your needs: such as removing one of the two entries and setting the paths and names to whatever you prefer.
+. Afterwards, you will be able to download the arbitrary files to the location specified in the `firmware_path` variable.
+. Upload the files for the virtual secondary to the https://connect.ota.here.com[OTA Connect Portal].
+. Now go to the *Software versions* tab and click *Add Software*. You should make sure the *ECU Types* field matches the value specified in the `ecu_hardware_id` variable of the `virtualsec.json` file.
++
+NOTE: `ecu_hardware_id` must be unique per each file type that you'll like to deliver, otherwise an update of one of them will rewrite the others.

```

Comment:
I understand this, but I'm not sure people new to the system will. Do we have a explanation somewhere that we can link to that describes why it works this way?

Reply:
@patrickvacek unfortunately we don't have extra information in our current documentation that explains this further. Maybe I could add the extra information here within the Note. 
Do you have any extra information on why it works this way which I could possibly add to the note to make it clearer for new users?

Reply:
I think we can find a better place to describe that topic, and that can probably wait for another PR. I loosely understand conceptually why the system works that way, but in my opinion it is non-obvious, confusing, and possibly just plain wrong. I forget if the web UI at least says something about it now or not. In any case, you'd probably need to talk to a backend engineer or the product team to explain why it works that way.

Reply:
@patrickvacek no problem. It was an extra point added by Mike within the description. 
Because this point is not yet fully explained in the doc, should I remove this note or do you think it is important to maintain the note for now and later write a more in-depth explanation on this?

Reply:
I think it's fine for now how you have it. Thanks!